bgpd/bgp_evpn.c

```diff
@@ -3037,6 +3037,14 @@ static int install_evpn_route_entry_in_vrf(struct bgp *bgp_vrf,
 			vrf_id_to_name(bgp_vrf->vrf_id), evp, parent_pi,
 			parent_pi->flags);
 
+	if (bgp_vrf->vrf_id == VRF_UNKNOWN) {
+		if (bgp_debug_zebra(NULL))
+			zlog_debug(
+			"vrf %s: vrf_id not set, evpn route not installed",
```

Comment:
`vrf %s` not needed, because `pretty_name` includes `VRF` or `view` itself.

Reply:
Yes, also if possible can we know the EVPN route to be printed also? Otherwise, it will be a mess when handling millions of routes for this condition?

Reply:
After some more digging I think maybe it would be better to treat it as an error instead of just additional log, so it would look like this:

```
	if (bgp_vrf->vrf_id == VRF_UNKNOWN) {
		flog_err(
			EC_BGP_EVPN_FAIL,
			"Failed to install EVPN %pFX route in %s, because vrf_id isn't set",
			evp, bgp_vrf->name_pretty);
		return -1;
	}
```

What do you think about that? Any other info that you would like to add into the error log?

Reply:
Actually after some more digging I can see that all 2 usages of this modified function:
```
		if (install)
			ret = install_evpn_route_entry_in_vrf(bgp_vrf, evp, pi);
		else
			ret = uninstall_evpn_route_entry_in_vrf(bgp_vrf, evp,
								pi);

		if (ret)
			flog_err(EC_BGP_EVPN_FAIL,
				 "Failed to %s EVPN %pFX route in VRF %s",
				 install ? "install" : "uninstall", evp,
				 vrf_id_to_name(bgp_vrf->vrf_id));
```

and 

```
		if (install)
			ret = install_evpn_route_entry_in_vrf(bgp_vrf, evp, pi);
		else
			ret = uninstall_evpn_route_entry_in_vrf(bgp_vrf, evp,
								pi);

		if (ret) {
			flog_err(EC_BGP_EVPN_FAIL,
				 "%u: Failed to %s prefix %pFX in VRF %s",
				 bgp_def->vrf_id,
				 install ? "install" : "uninstall", evp,
				 vrf_id_to_name(bgp_vrf->vrf_id));
			return ret;
		}
```

Have an error log after it is called that logs exactly what would happen in this case, like which prefix failed to install and in which VRF. Do you think we should add more details to install_evpn_route_entry_in_vrf anyways? If so, should I log it as zlog_debug() like:

```
	if (bgp_vrf->vrf_id == VRF_UNKNOWN) {
	        if (bgp_debug_zebra(NULL))
		        zlog_debug(
			        "%s, failed to install EVPN %pFX route in because vrf_id isn't set",
 			        bgp_vrf->name_pretty, evp);
		return -1;
	}
```

or flog_err() like:

```
	if (bgp_vrf->vrf_id == VRF_UNKNOWN) {
		flog_err(
			EC_BGP_EVPN_FAIL,
			"Failed to install EVPN %pFX route in %s, because vrf_id isn't set",
			evp, bgp_vrf->name_pretty);
		return -1;
	}
```

Reply:
Correct, logging is not needed at all then, missed that the return code is checked up the path.

Reply:
Ok - logs removed.